specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-04-01-preview/Hunts.json

```diff
@@ -0,0 +1,1001 @@
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "hypothesisStatus"
+          },
+          "default": "Unknown"
+        },
+        "attackTactics": {
+          "description": "A list of mitre attack tactics the hunt is associated with",
+          "items": {
+            "$ref": "./common/AlertTypes.json#/definitions/AttackTactic"
+          },
+          "type": "array"
+        },
+        "attackTechniques": {
+          "description": "A list of a mitre attack techniques the hunt is associated with",
+          "items": {
+            "description": "Attack Technique",
+            "type": "string"
+          },
+          "type": "array"
+        },
+        "labels": {
+          "description": "List of labels relevant to this hunt ",
+          "items": {
+            "$ref": "../../../common/2.0/types.json#/definitions/Label"
+          },
+          "type": "array"
+        },
+        "owner": {
+          "$ref": "#/definitions/HuntOwner"
+        }
+      },
+      "required": [
+        "displayName",
+        "description"
+      ],
+      "type": "object"
+    },
+    "HuntRelationList": {
+      "description": "List of all the hunt relations",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of hunt relations.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of hunt relations",
+          "items": {
+            "$ref": "#/definitions/HuntRelation"
+          },
+          "type": "array"
+        }
+      },
+      "type": "object",
+      "required": [
+        "value"
+      ]
+    },
+    "HuntRelation": {
+      "allOf": [
+        {
+          "$ref": "../../../common/2.0/types.json#/definitions/ResourceWithEtag"
+        }
+      ],
+      "description": "Represents a Hunt Relation in Azure Security Insights.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/HuntRelationProperties",
+          "description": "Hunt Relation properties",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object"
+    },
+    "HuntRelationProperties": {
+      "description": "Describes hunt relation properties",
+      "properties": {
+        "relatedResourceId": {
+          "description": "The id of the related resource",
+          "type": "string"
+        },
+        "relatedResourceName": {
+          "description": "The name of the related resource",
+          "type": "string"
+        },
+        "relationType": {
+          "description": "The type of the hunt relation",
+          "type": "string"
+        },
+        "relatedResourceKind": {
+          "description": "The resource that the relation is related to",
+          "enum": [
+            "Hunt",
+            "Incident",
+            "Bookmark",
+            "Entities",
+            "AmlWorkspace",
+            "LogAnalyticsSavedSearchQueries"
```

Comment:
why do you need the name type and kind if you are already getting the full resource id of this resource ? It is recommended to avoid asking more data from the user than that is absolutely necessary in such apis , you run into issues of what if these mismatch with the relatedREsourceId value and also is cumbersome for user to craft

Reply:
That is a valid point. However, our API performs checks if there is a mismatch, and we need to validate whether the resource type/kind is supported. RelatedResourceKind is not required, but RelatedResourceId is, which we are extracting the Kind from.
I believe looking back, we had this design of having type and kind because we had to match our API design with an existing feature.
https://github.com/Azure/azure-rest-api-specs/blob/main/specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2023-03-01-preview/common/RelationTypes.json

Reply:
You can extract name and type from the id as well . Can you fix this now in this iteration ?

Reply:
I have gathered more clarification and have made updates to the properties. readOnly has been applied to name, kind, and type.

As for type, we need to specify which provider the resource is coming from: either Microsoft.SecurityInsights or Microsoft.OperationalInsights.

As for kind, we have removed the enum aspect. Kind is not the same as type and is only populated when the type is a specific value: Entities. Kind can be a value from here or more: https://learn.microsoft.com/en-us/rest/api/securityinsights/preview/entities/get?tabs=HTTP#examples